### PR TITLE
HPCC-20587 esdl2ecl default generated ECL strings to UTF8

### PR DESCRIPTION
**Pull request by afishbeck**
Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by afishbeck** [comment]
@ghalliday This is the switch to default to using utf8 for ESDL generated query interfaces.
@rpastrana please review

**Comment by AttilaVamos** [comment]
@afishbeck The missing part of the build error is:
```
Run CPack packaging tool...
CPack: Create package using RPM
CPack: Install projects
CPack: - Run preinstall target for: hpccsystems-platform
CPack: - Install project: hpccsystems-platform
CMake Error at /mnt/disk1/home/vamosax/smoketest/PR-11743/build/plugins/couchbase/cmake_install.cmake:60 (file):
  file INSTALL cannot find
  "/mnt/disk1/home/vamosax/smoketest/PR-11743/build/plugins/couchbase/libcouchbase-build/lib/libcouchbase.so.2.0.59".
Call Stack (most recent call first):
  /mnt/disk1/home/vamosax/smoketest/PR-11743/build/plugins/cmake_install.cmake:61 (include)
  /mnt/disk1/home/vamosax/smoketest/PR-11743/build/cmake_install.cmake:51 (include)

CPack Error: Error when generating package: hpccsystems-platform
```

File: tools/esdlcmd/esdl2ecl.cpp [hunk 1]
```diff
@@ -269,6 +269,8 @@ class Esdl2EclCmd : public EsdlConvertCmd
             return EsdlCmdOptionMatch;
```
Comment (rpastrana) [inline tools/esdlcmd/esdl2ecl.cpp]: was the change to libcouchbase above intentional?

File: tools/esdlcmd/esdl2ecl.cpp [hunk 2]
```diff
@@ -389,6 +391,7 @@ class Esdl2EclCmd : public EsdlConvertCmd
                 "      --ecl-imports         Comma-delimited import list to be attached to output ECL\n"
                 "                            each entry generates a corresponding import *.<entry>\n"
                 "      --ecl-header          Text included in target header (must be valid ECL) \n"
+                "      --utf8-               Don't use UTF8 strings"
```
Comment (rpastrana) [inline tools/esdlcmd/esdl2ecl.cpp]: I assume the "-" postfix is the notation to negate the option?

File: esp/xslt/esdl2ecl.xslt [hunk 3]
```diff
@@ -239,6 +240,13 @@ END;
 
 </xsl:template>
 
+<xsl:template name="output_ecl_stringtype">
+  <xsl:choose>
+    <xsl:when test="$utf8strings='yes'">UTF8</xsl:when>
```
Comment (rpastrana) [inline esp/xslt/esdl2ecl.xslt]: it looks like the rest of the xsl wraps text in <xsl:text disable-output-escaping="yes">, do we need to do that here?

**Review by rpastrana — COMMENTED** [review]
@afishbeck just a few comments

File: esp/xslt/esdl2ecl.xslt (hunk 3 above)
Comment (afishbeck) [reply esp/xslt/esdl2ecl.xslt]: In this case I know nothing will be escaped.  The characters are all ascii.

File: tools/esdlcmd/esdl2ecl.cpp (hunk 2 above)
Comment (afishbeck) [reply tools/esdlcmd/esdl2ecl.cpp]: Yes.  In reality I think it should always be UTF8 so this notation is only meant if for some reason someone wants to turn it off.  I hope no one needs to.. but maybe during the transition.

File: tools/esdlcmd/esdl2ecl.cpp (hunk 1 above)
Comment (afishbeck) [reply tools/esdlcmd/esdl2ecl.cpp]: ugh. unintentional. I'll take a look.

Comment (afishbeck) [reply tools/esdlcmd/esdl2ecl.cpp]: removed

**Comment by afishbeck** [comment]
@rpastrana commented and updated without libcouchbase issue.

File: tools/esdlcmd/esdl2ecl.cpp (hunk 2 above)
Comment (afishbeck) [reply tools/esdlcmd/esdl2ecl.cpp]: By the way the postfix is just the standard used in Gavins parameter processor.  The option works without, or with plus, but both of those are the default, so I just show the one way that actually changes the default.

**Review by rpastrana — APPROVED** [review]
@afishbeck looks reasonable

**Pull request merged**